File: mc/rcmd/src/main/java/calebxzhou/rdi/mc/rcmd/RcmdResult.java
package calebxzhou.rdi.mc.rcmd;

public record RcmdResult(boolean success, String message) {
    public RcmdResult {
        message = message == null ? "" : message;
    }

    public static RcmdResult ok() {
        return new RcmdResult(true, "");
    }

    public static RcmdResult ok(String message) {
        return new RcmdResult(true, message);
    }

    public static RcmdResult error(String message) {
        return new RcmdResult(false, message);
    }
}

File: mc/rcmd/src/main/kotlin/calebxzhou/rdi/mc/rcmd/RcmdResult.kt
package calebxzhou.rdi.mc.rcmd

data class RcmdResult(val success: Boolean, val message: String) {
    fun success() = success

    fun message() = message

    companion object {
        @JvmStatic
        fun ok() = RcmdResult(true, "")

        @JvmStatic
        fun ok(message: String) = RcmdResult(true, message)

        @JvmStatic
        fun error(message: String?) = RcmdResult(false, message.orEmpty())
    }
}
